src/main/java/com/checkmarx/flow/service/FlowService.java

```diff
@@ -152,33 +152,56 @@ public ScanDetails executeCxScan(ScanRequest request, File cxFile) throws Machin
                     projectId = cxService.getProjectId(ownerId, projectName); //get the project id of the updated or created project
                 }
                 osaScanId = createOsaScan(request, projectId);
-                logRequest(request, osaScanId);
+                if(osaScanId != null) {
+                    logRequest(request, osaScanId, cxFile, Status.SUCCESS);
+                }
                 //resultsService.processScanResultsAsync(request, projectId, scanId, osaScanId, request.getFilters());
             }
 
 
         }catch (CheckmarxException | GitAPIException e){
             log.error(ExceptionUtils.getMessage(e), e);
             Thread.currentThread().interrupt();
-            new ScanReport(scanId, request, sourcesPath, Status.FAILURE.build(e.getMessage())).log();
+            logRequest(request, scanId, cxFile, Status.FAILURE.build(e.getMessage()));
```

Comment:
In case of a failure, don't you prefer to print out the whole stack-trace (e) instead only the message?